test/release/examples/primers/distributions.chpl

```diff
@@ -1,354 +1,350 @@
+// argument, as we'll see below.
 //
 
 //
-// Make the program size configurable from the command line.
+// First, we'll declare the problem size to use for our 2D domains
+// and arrays, configurable from the command-line:
 //
 config const n = 8;
 
 //
-// Declare a 2-dimensional domain ``Space`` that we will later use to
-// initialize the distributed domains.
+// Next, we'll declare a 2-dimensional domain ``Space``, which will
+// form the basis of our distributed domains.
 //
 const Space = {1..n, 1..n};
 
-// blockDist (and distribution basics)
-// -----------------------------------
+// The Block Distribution
+// ----------------------
 //
-// The ``blockDist`` distribution partitions an n-dimensional bounding
-// box between a set of target locales arranged into a conceptual
-// n-dimensional grid.  The bounding box is divided into roughly
-// equal-size block sections, where each locale owns one of them.
-// Note that domains declared using a ``blockDist`` distribution can
-// also store indices outside of the bounding box; the bounding box is
-// merely used to compute a partitioning of n-dimensional space, not
-// to constrain legal domains.
+// The ``blockDist`` distribution partitions a *d*-dimensional
+// bounding box across the *d*-dimensional target locale array.  The
+// bounding box is divided into roughly equal-size block sections,
+// where each locale owns one of them.
+
 //
 // In this example, we declare a 2-dimensional block distribution
-// whose bounding box is defined by ``Space``, a domain over that
-// distribution using the same index set, and an array over the
-// domain.
+// whose bounding box is defined by ``Space``.  We then use that
+// distribution instance to create a distributed domain and an array
+// over the domain:
 //
 const BlkDist = new blockDist(boundingBox=Space);
 const BlockSpace = BlkDist.createDomain(Space);
 var BA: [BlockSpace] int;
 
 /*
-The above could also be shortened to one of the following for convenience:
 
-.. code-block:: chapel
+  Since the code above only uses a single domain and array, it could
+  also be written in one of the following ways for convenience:
 
-	const BlockSpace = blockDist.createDomain(Space);  
-	const BA: [BlockSpace] int;
+  .. code-block:: chapel
 
-or:
+    // create the domain using an anonymous distribution
+    const BlockSpace = blockDist.createDomain({1..n, 1..n});
+    const BA: [BlockSpace] int;
 
-.. code-block:: chapel
+  or:
+
+  .. code-block:: chapel
 
-	const BA = blockDist.createArray(Space);
+    // create the array using an anonymous distribution and domain
+    const BA = blockDist.createArray({1..n, 1..n}, int);
+
+  Or, in both versions, `Space` could be substituted for the domain
+  literal, if preferred.
+
+  In this primer, we'll be illustrating the creation of multiple
+  aligned domains, which is only possible by declaring the
+  distribution as a distinct variable.  And in real programs, it can
```

Comment:
@vasslitvinov: Good point.  I was trying to rationalize why the primer doesn't use these shorthands, but ended up doing it in a fairly roundabout way.  How about:

>   One motivation for declaring explicit distribution and domain
  variables as we've done here is to support the creation of aligned
  domains and arrays.  Another is to amortize the overheads incurred
  by such types, since creating any distributed object requires
  communication and memory.

Reply:
👍 works for me!